code-challenges.test.js

```diff
@@ -37,15 +88,112 @@ const votes1 = { upVotes: 13, downVotes: 2 }
+    })
+  })
+
+//Test failed as expected: ReferenceError: finalTally is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 passed, 2 total
 
 // b) Create the function that makes the test pass.
 
+//Pseudo: 
+//Create a function called finalTally
+//Input: An object
+//Destructure object to be able to access values easier without having to use dot notation.
+  //Do subtraction on the upvotes and downvotes and store into a variable
+//Output: The final tally based on up and down votes. 
+
+const finalTally = (object) => {
+    const {upVotes, downVotes} = object   //Object destructuring 
+    const totalVotes = upVotes - downVotes
+  return totalVotes
+}
+
+//Test logs to check if key-pair values deconstructed properly and subtraction outputed correctly. 
+console.log(finalTally(votes1))
+console.log(finalTally(votes2))
+//Test log to check for different values
+console.log(finalTally(votes3))
+
+
+//Test passed as expected
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
 const dataArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+//Test variable to check if the spread operator would work correctly. 
+const dataArray3 = [5, 6, 9, 81, 67, 9]
+//Output:   ['array', 'object', 'number', 'string', 'Boolean', 'null', 'undefined', 5, 6, 9, 81, 67]
+
+
+describe("noDuplicates", () => {
+  it("returns an array with no duplicate values", () => {
+    expect(noDuplicates(dataArray1)).toEqual(expect.not.arrayContaining(["array", "object", "number", "string", "Boolean", "null", "undefined"]))
+    expect(noDuplicates(dataArray3)).toEqual(expect.not.arrayContaining(['array', 'object', 'number', 'string', 'Boolean', 'null', 'undefined', 5, 6, 9, 81, 67]))
+  })
+})
+
+//Test failed as expected: ReferenceError: noDuplicates is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 2 passed, 3 total
 
 // b) Create the function that makes the test pass.
+
+// Pseudo: Original function
+//Create a function called noDuplicates, using an array as the input. 
+  //Combine both arrays by using .concat and store result into a variable.
+  //Use set method and pass in the combined array to create a set of unique values(essentially eliminating duplicate values). 
+    //Each value in a set can only occur once. 
+//Output: One array with no duplicate values. 
+
+//Original function using .concat and Set
+// const noDuplicates = (array) => {              
+//   const combinedArray = array1.concat(array2)
+//   return [...new Set (combinedArraay)]
```

Comment:
I believe it could have possibly been the typo on line 168 with combinedArray. It has an extra a but this lead to a really good refactor.